sysdeps/managarm/generic/file.cpp

```diff
@@ -2412,6 +2436,8 @@ int sys_ioctl(int fd, unsigned long request, void *arg, int *result) {
 
 		HEL_CHECK(offer->error);
 		HEL_CHECK(send_req->error);
+		if(recv_resp->error == kHelErrDismissed)
+			return EINVAL;
 		HEL_CHECK(send_mode->error);
```

Comment:
Looks like this should be `send_mode->error` (if that's the first item after the request)?